openvdb_ax/openvdb_ax/cmd/openvdb_ax.cc

```diff
@@ -462,26 +576,37 @@ main(int argc, char *argv[])
     ScopedInitialize initializer(argc, argv);
     axlog(": " << axtime() << '\n');
 
-    // read vdb file data for
+    std::unique_ptr<tbb::global_control> control;
+    if (opts.threads > 0) {
+        axlog("[INFO] Initializing thread usage [" << opts.threads << "]\n" << std::flush);
+        control.reset(new tbb::global_control(tbb::global_control::max_allowed_parallelism, opts.threads));
+    }
 
-    openvdb::GridPtrVecPtr grids;
+    openvdb::GridPtrVec grids;
     openvdb::MetaMap::Ptr meta;
 
     if (opts.mMode == ProgOptions::Execute) {
-        openvdb::io::File file(opts.mInputVDBFile);
-        try {
-            axtimer();
-            axlog("[INFO] Reading VDB data"
-                << (openvdb::io::Archive::isDelayedLoadingEnabled() ?
-                    " (delay-load)" : "") << std::flush);
-            file.open();
-            grids = file.getGrids();
-            meta = file.getMetadata();
-            file.close();
-            axlog(": " << axtime() << '\n');
-        } catch (openvdb::Exception& e) {
-            OPENVDB_LOG_ERROR(e.what() << " (" << opts.mInputVDBFile << ")");
-            return EXIT_FAILURE;
+        // read vdb file data for
+        axlog("[INFO] Reading VDB data"
+            << (openvdb::io::Archive::isDelayedLoadingEnabled() ?
+                " (delay-load)" : "") << '\n');
+        for (const auto& filename : opts.mInputVDBFiles) {
+            openvdb::io::File file(filename);
+            try {
+                axlog("[INFO] | \"" << filename << "\"");
+                axtimer();
+                file.open();
+                auto in = file.getGrids();
+                grids.insert(grids.end(), in->begin(), in->end());
+                // choose the first files metadata
+                if (!meta) meta = file.getMetadata();
```

Comment:
maybe the metadata should combine all the metadatas? take the first version of entry?

Reply:
should also document this somewhere

Reply:
You run into a new set of problems with combining them regarding duplicate entries. We should add options for this down the line. Perhaps we don't even copy any file level metadata and just output a new file? That might make the most sense but not sure what this was doing before

Reply:
Yeah sorry I meant take the first version of each entry that occurs, to handle duplicates. Maybe as you say if before it would not copy the metadata, we just don't do that for now? Not sure what the options would be apart from first or last version of each entry.

Reply:
So the old behaviour was to copy the metadata of the input file which is what this currently does. Probably makes more sense not to do that (for example file metadata from Houdini has the "creator" info in it, it's most likely not expected to be propagated)

Reply:
I've added this option and removed the file meta copy by default